firefox_ui_harness/testcases/update.py

```diff
@@ -0,0 +1,133 @@
+            }
+            self.software_update.update_channel.default_channel = self.update_channel
+
+        # If requested modify the list of allowed MAR channels
+        # TODO: Bug 1142805 - Modify file via Python directly
+        if self.update_mar_channels:
+            # Backup the original content and the path of the update-settings.ini file
+            self.default_mar_channels = {
+                'content': self.software_update.mar_channels.config_file_contents,
+                'path': self.software_update.mar_channels.config_file_path,
+            }
+            self.software_update.mar_channels.add_channels(self.update_mar_channels)
+
+        # Bug 1142805 - Until we don't modify the channel-prefs.js and update-settings.ini
+        # files before Firefox gets started, a restart of Firefox is necessary to
+        # accept the new update channel.
+        self.restart()
+
+        # Dictionary which holds the information for each update
+        self.updates = [{
+            'build_pre': self.software_update.build_info,
+            'build_post': self.software_update.build_info,
+            'fallback': is_fallback,
+            'patch': {},
+            'success': False,
+        }]
+
+        self.assertEqual(self.software_update.update_channel.default_channel,
+                         self.software_update.update_channel.channel)
+
+        self.assertTrue(self.update_mar_channels.issubset(
+                        self.software_update.mar_channels.channels))
+
+    def assert_update_applied(self, update_status):
+        """Checks if an update has been applied correctly.
+
+        :param update_status: All the data collected during the update process
+        """
+        # Get the information from the last update
+        info = update_status[self.current_update_index]
+
+        # The upgraded version should be identical with the version given by
+        # the update and we shouldn't have run a downgrade
+        check = self.marionette.execute_script("""
+          Components.utils.import("resource://gre/modules/Services.jsm");
+
+          return  Services.vc.compare(arguments[0], arguments[1]);
+        """, script_args=[info['build_post']['version'], info['build_pre']['version']])
+
+        self.assertGreaterEqual(check, 0, 'The version of the upgraded build is higher or equal')
+
+        # If a target version has been specified, check if it matches the updated build
+        if self.target_version:
+            self.assertEqual(info['build_post']['version'], self.target_version)
+
+        # The post buildid should be identical with the buildid contained in the patch
+        self.assertEqual(info['build_post']['buildid'], info['patch']['buildid'])
+
+        # If a target buildid has been specified, check if it matches the updated build
+        if self.target_buildid:
+            self.assertEqual(info['build_post']['buildid'], self.target_buildid)
```

Comment:
There is no need for that. See line 91 where we already assert that the post build id has to be equal to the build id of the downloaded patch. That is the reference we need.